gobblin-runtime/src/test/java/org/apache/gobblin/runtime/spec_store/MysqlBaseSpecStoreTest.java

```diff
@@ -116,6 +135,24 @@ public void testAddSpec() throws Exception {
     Assert.assertFalse(this.specStore.exists(URI.create("dummy")));
   }
 
+  @Test
+  public void testUpdateSpec() throws Exception {
+    this.specStore.addSpec(this.flowSpec);
+    Assert.assertEquals(this.specStore.getSize(), 1);
+    Assert.assertTrue(this.specStore.exists(this.uri1));
+    Assert.assertFalse(this.specStore.exists(URI.create("dummy")));
+
+    //Updating
+
+    this.specStore.addSpec(this.flowSpecUpdated);
+    Assert.assertEquals(this.specStore.getSize(), 1);
+    Assert.assertTrue(this.specStore.exists(this.uri1));
+    FlowSpec updated = (FlowSpec)this.specStore.getSpecImpl(this.uri1);
+    Assert.assertEquals(updated.getConfigAsProperties().getProperty("key"),"valueUpdated");
+    Assert.assertEquals(updated.getConfigAsProperties().getProperty("key3"),"value3UPdated");
+    Assert.assertEquals(updated.getConfigAsProperties().getProperty("config.with.dot"),"value4Updated");
```

Comment:
none of these verify any of the special "indexing" columns you specified in the `ON DUPLICATE KEY` clause, such as `user_to_proxy`, `source_identifier`, etc.

being the crux of this PR, that would seem worth adding

Reply:
addressed